src/remote/integration.ts

```diff
@@ -50,42 +30,44 @@ function remoteExtensionsInstalled(): boolean {
 export async function addRemoteIntegration(context: vscode.ExtensionContext) {
   State.extensionContext = context;
 
-  registerRemoteIntegrationCommands();
-
   const remoteExtensions = remoteExtensionsInstalled();
   await vscode.commands.executeCommand('setContext', 'peacock:remote', remoteExtensions);
 
-  if (!vscode.env.remoteName) {
-    revertRemoteWorkspaceColors();
-    return;
-  }
-  // don't overwrite an already defined custom color
-  if (!getPeacockColorWorkspaceMemento()) {
-    setRemoteWorkspaceColors();
-  }
-}
+  // TODO: Shouldnt need this, since changeColor now changes the color for remote or non-remote
+  // if (!vscode.env.remoteName) {
+  //   revertRemoteWorkspaceColors();
+  //   return;
+  // }
 
-export async function refreshRemoteColor(remote: string): Promise<boolean> {
-  if (vscode.env.remoteName !== remote) {
-    notify(
-      `The selected color will be applied every time you you are in the '${remote}' context.`,
-      true,
-    );
-    return false;
+  if (vscode.env.remoteName) {
+    await setRemoteWorkspaceColors();
```

Comment:
The call to `setRemoteWorkspaceColors()` will have the side effect to call `updatePeacockRemoteColor`, which changes the setting value. On this code path the setting value should only be consumed but not updated.